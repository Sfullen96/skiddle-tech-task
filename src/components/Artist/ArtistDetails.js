import React from 'react';
import PropTypes from 'prop-types';
import { Container, Row, Col } from 'react-bootstrap';

import './Artist.scss';

const ArtistDetails = ({ name, image, twitter, description, audioPreview }) => (
  <Container className="artist">
    <Row>
      <Col xs="12" sm="3" className="event__image-container">
        <img
          src={image}
          alt={`${name}`}
          className="figure-img img-fluid img-thumbnail artist__image"
        />
      </Col>
      <Col xs="12" sm="9" className="event__main-info">
        <Row>
          <Col xs="12" md="8">
            <h1 className="artist__name">{name}</h1>
            {twitter && (
              <p className="artist__twitter">
                <i className="fab fa-twitter" />{' '}
                <a
                  href={`https://twitter.com/${twitter}`}
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  {`https://twitter.com/${twitter}`}
                </a>
              </p>
            )}
          </Col>
          <Col xs="12" md="4">
            {/* disable the warning about not having a <track /> as one was not provided for this test */}
            {/* eslint-disable-next-line */}
            <audio controls className="artist__preview">
              <source src={audioPreview} type="audio/mpeg" />
              Your browser does not support the audio element.
            </audio>
          </Col>
        </Row>
        <div className="artist__description">
          <p>{description}</p>
        </div>
      </Col>
    </Row>
  </Container>
);

ArtistDetails.defaultProps = {
  twitter: null,
};

ArtistDetails.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  twitter: PropTypes.string,
  description: PropTypes.string.isRequired,
  audioPreview: PropTypes.string.isRequired,
};

export default ArtistDetails;
