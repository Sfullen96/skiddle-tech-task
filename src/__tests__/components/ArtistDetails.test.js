import React from 'react';
import { shallow } from 'enzyme';

import ArtistDetails from '../../components/Artist/ArtistDetails';

describe('<ArtistDetails />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <ArtistDetails
        name="Courteeners"
        image="testimage.png"
        description="test"
        audioPreview="audio.mp3"
      />,
    );
  });

  it('should match the snaphot', () => expect(wrapper).toMatchSnapshot());

  it('should render without crashing', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('should render the artist name', () => {
    expect(wrapper.find('.artist__name').text()).toEqual('Courteeners');
  });

  it('should render the image of the artist, and it should have an appropriate alt', () => {
    wrapper.setProps({ image: 'testimage.png', name: 'Test Artist' });
    expect(wrapper.find('.artist__image').props().src).toEqual('testimage.png');
    expect(wrapper.find('.artist__image').props().alt).toEqual('Test Artist');
  });

  it('should not render a twitter link if the prop is not available', () => {
    wrapper.setProps({ twitter: null });
    expect(wrapper.find('artist__twitter').length).toEqual(0);
  });

  it('should render a twitter link', () => {
    wrapper.setProps({ twitter: 'thecourteeners' });
    expect(wrapper.find('.artist__twitter').containsAllMatchingElements([<p />, <a />, <i />]));
    expect(
      wrapper
        .find('.artist__twitter')
        .find('a')
        .props().href,
    ).toEqual('https://twitter.com/thecourteeners');
    expect(
      wrapper
        .find('.artist__twitter')
        .find('a')
        .text(),
    ).toEqual('https://twitter.com/thecourteeners');
  });

  it('should render the description of the artist', () => {
    expect(wrapper.find('.artist__description p').text()).toEqual('test');
  });

  it('should render an audio preview', () => {
    expect(wrapper.find('.artist__preview source').props().src).toEqual('audio.mp3');
  });
});
