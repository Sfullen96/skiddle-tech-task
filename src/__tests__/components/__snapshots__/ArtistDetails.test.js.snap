// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArtistDetails /> should match the snaphot 1`] = `ShallowWrapper {}`;
